include/gridtools/common/defs.hpp

```diff
@@ -27,13 +27,17 @@ namespace gridtools {
 #endif
 #endif
 
-#if defined(__has_builtin)
+#ifdef __cpp_attributes
+#if __has_cpp_attribute(assume)
+#define GT_ASSUME(x) [[assume(x)]]
+#endif
+#endif
+#if !defined(GT_ASSUME) && defined(__has_builtin)
```

Comment:
Wondering if we should trust nvcc that `[[assume(x)]] == __builtin_assume(x)` once they introduce it.

Reply:
Good point. On the one hand, it would be the only reasonable thing, but on the other hand it’s NVCC.
I just wanted to try with CUDA 12.5, but there seems to be another problem: `__has_builtin(__builtin_assume)` actually returns zero, so our `__builtin_assume` actually never seems to be enabled with NVCC.
I’ll modify this PR a bit and add a test to make sure `__builtin_assume` is activated on NVCC.